### PR TITLE
Fix failing test in worker spec

### PR DESCRIPTION
**Pull request by sonianand11**
This PR includes
- use of present? method in `is_requesting?` method of worker
- modify check for requesting in worker spec `returns true if there is a current batch`

@benjaminhoffman 

Note : Intention of this PR is to re run test in travis

### CODE REVIEW

File: lib/segment/analytics/worker.rb [hunk 1]
```diff
@@ -41,19 +41,17 @@ def run
...
-
           @lock.synchronize { @batch.clear }
         end
       end
 
       # public: Check whether we have outstanding requests.
       #
       def is_requesting?
-        @lock.synchronize { !@batch.empty? }
+        @lock.synchronize { @batch.present? }
```
Comment (nettofarah) [inline lib/segment/analytics/worker.rb]: this relies on activesupport, which is a dependency you usually don't want to have in more generic gems.
We might wanna change this to use `any?`

File: lib/segment/analytics/worker.rb [hunk 2]
```diff
@@ -41,19 +41,17 @@ def run
               @batch << @queue.pop
             end
           end
-
           res = Request.new.post @write_key, @batch
```
Comment (nettofarah) [inline lib/segment/analytics/worker.rb]: no need to remove this whitespace

File: spec/segment/analytics/worker_spec.rb [hunk 3]
```diff
@@ -85,16 +85,18 @@ class Analytics
         end
 
         it 'returns true if there is a current batch' do
+
```
Comment (nettofarah) [inline spec/segment/analytics/worker_spec.rb]: no need to add this extra line

File: lib/segment/analytics/worker.rb (hunk 2 above)
Comment (nettofarah) [inline lib/segment/analytics/worker.rb]: I meant *new line.

File: lib/segment/analytics/worker.rb (hunk 1 above)
Comment (sonianand11) [inline lib/segment/analytics/worker.rb]: I used it as we already have it in our gemspec.

**Pull request merged**